pkg/utils/utils.go

```diff
@@ -538,28 +547,20 @@ func GetPodUserId(pod *v1.Pod) (int64, int64) {
 			// No need to handle the case for the initContainers here
 			// since it is a different list in pod spec i.e. pod.Spec.InitContainers
 			securityContext := container.SecurityContext
-			uid, gid = getIdFromSecurityContext(securityContext)
-			if uid != UndefinedId || gid != UndefinedId {
-				return uid, gid
-			}
+			containerUid, containerGid = getIdFromSecurityContext(securityContext)
```

Comment:
Yes we should have broken from the loop. 
But as per our discussion let me add the logic of getting the UID and GID in whichever container it is set. Not just the first one. It may happen that in an ore. than one container scenario 3rd or 4th container has set the security context then we will loose it completely , hence we can implement that logic here.
But if each container has a different securityContext defined in its own setup then it is not handle today. Infact if you think all the DBs none of them have such a security context set.
So lets agree top go with this approach now.